mm/mm_heap/mm_malloc.c

```diff
@@ -114,11 +114,11 @@ FAR void *mm_malloc(FAR struct mm_heap_s *heap, size_t size)
 
   free_delaylist(heap);
 
-  /* Ignore zero-length allocations */
+  /* Ensure we request an allocation of at least one byte */
 
   if (size < 1)
     {
-      return NULL;
+      size = 1;
     }
```

Comment:
This entire conditional is unnecessary and should be removed.  The purpose was not to set the minimum allocation to one, but rather to force the minimum chunk size.  That that will happen anyway when size == 0.  See https://github.com/apache/nuttx/blob/master/mm/mm_heap/mm_malloc.c#L137

So all of the code at line 117 can be removed:

    -  /* Ignore zero-length allocations */
    -
    -  if (size < 1)
    -    {
    -      return NULL;
    -    }